spring-beans/src/main/java/org/springframework/beans/factory/annotation/InjectionMetadata.java

```diff
@@ -107,15 +107,20 @@ protected boolean needsRefresh(Class<?> clazz) {
 	}
 
 	public void checkConfigMembers(RootBeanDefinition beanDefinition) {
-		Set<InjectedElement> checkedElements = new LinkedHashSet<>(this.injectedElements.size());
-		for (InjectedElement element : this.injectedElements) {
-			Member member = element.getMember();
-			if (!beanDefinition.isExternallyManagedConfigMember(member)) {
-				beanDefinition.registerExternallyManagedConfigMember(member);
-				checkedElements.add(element);
+		if (this.injectedElements.isEmpty()) {
+			this.checkedElements = Set.of();
+		}
+		else {
+			Set<InjectedElement> checkedElements = new LinkedHashSet<>(this.injectedElements.size());
```

Comment:
This should be size*4/3 to avoid hash resizes in the for loop below

Reply:
See https://richardstartin.github.io/posts/5-java-mundane-performance-tricks

Reply:
Fixed